Common/File/FileUtil.cpp

```diff
@@ -97,6 +100,48 @@ FILE *OpenCFile(const Path &path, const char *mode) {
 	switch (path.Type()) {
 	case PathType::NATIVE:
 		break;
+	case PathType::CONTENT_URI:
+		// We're gonna need some error codes..
+		if (!strcmp(mode, "r") || !strcmp(mode, "rb")) {
+			INFO_LOG(COMMON, "Opening content file for read: '%s'", path.c_str());
+			// Read, let's support this - easy one.
+			int descriptor = Android_OpenContentUriFd(path.ToString(), Android_OpenContentUriMode::READ);
+			if (descriptor == -1) {
+				return nullptr;
+			}
+			return fdopen(descriptor, "rb");
+		} else if (!strcmp(mode, "w") || !strcmp(mode, "wb")) {
+			// Need to be able to create the file here if it doesn't exist.
+			// Not exactly sure which abstractions are best, let's start simple.
+			if (!File::Exists(path)) {
+				INFO_LOG(COMMON, "Opening content file '%s' for write. Doesn't exist, creating empty and reopening.", path.c_str());
+				std::string name = path.GetFilename();
+				if (path.CanNavigateUp()) {
+					Path parent = path.NavigateUp();
+					if (!Android_CreateFile(parent.ToString(), name)) {
+						WARN_LOG(COMMON, "Failed to create file '%s' in '%s'", name.c_str(), parent.c_str());
+						return nullptr;
+					}
+				} else {
+					INFO_LOG(COMMON, "Failed to navigate up to create file");
+					return nullptr;
+				}
+			} else {
+				INFO_LOG(COMMON, "Opening file by fd for write");
+			}
+
+			// Read, let's support this - easy one.
+			int descriptor = Android_OpenContentUriFd(path.ToString(), Android_OpenContentUriMode::READ_WRITE_TRUNCATE);
+			if (descriptor == -1) {
+				INFO_LOG(COMMON, "Opening '%s' for write failed", path.ToString().c_str());
+				return nullptr;
+			}
+			return fdopen(descriptor, "wb");
+		} else {
+			ERROR_LOG(COMMON, "OpenCFile(%s): Mode not yet supported: %s", path.c_str(), mode);
```

Comment:
REPORT?  I tend to use "rt" or "wt" sometimes, but I think we don't have that.  Could definitely see r+, w+, etc.

-[Unknown]

Reply:
Will add reporting after moving that machinery to Common.